### DISCUSSION TITLE
[Feature] Document UUID version, function for ULIDs and/or enable different versions of UUID

### DISCUSSION THREAD

**Discussion by mxcoder** [duckdb/duckdb#11047] [Ideas]
It would be great to have also an option to generate these different types of Unique IDs.

1. The docs for `uuid()` and `gen_random_uuid()` do not mention what version of UUID are generated.
2. [ULID](https://github.com/ulid/spec) can also be a quite useful unique ID format
3. [UUID v7](https://buildkite.com/blog/goodbye-integers-hello-uuids) is still in draft, but would be great to be ahead of the curve.

The general utility of using different formats of unique IDs depending of the use case could be a powerful feature of duckdb when processing chunks of data in a distributed way.

**Comment by sergeyprokhorenko** [comment]
See https://commitfest.postgresql.org/47/4388/
UUIDv7 was not included into the 17th version of PostgreSQL due to a feature freeze

**Comment by jc00ke** [comment]
I was pleasantly surprised to see a [community extension for ULID](https://duckdb.org/community_extensions/extensions/ulid).

**Comment by Mytherin** [comment]
We've merged in UUID v7 support here - https://github.com/duckdb/duckdb/pull/15819 - this will be supported in core in the next release
